agent/structs/config_entry_gateways.go

```diff
@@ -231,11 +239,49 @@ func (e *IngressGatewayConfigEntry) validateServiceSDS(lis IngressListener, svc
 	return nil
 }
 
+func validateGatewayTLSConfig(tlsCfg GatewayTLSConfig) error {
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
```

Comment:
Should we do anything with `tlsCfg.Enabled` here, either require the rest of the fields be empty, or skip validating them and let them be full of garbage?

Reply:
Could be possible but would require a bit more refactoring as this function is used to validate the `TLS` field on both the gateway and listeners.

For the listener case, we'd have to check against the value of `tlsCfg.Enabled` for the _gateway_, and not just the listener, as `false` is the zero-value of this struct field and won't override a value of `true` on the gateway itself (see discussion at https://github.com/hashicorp/consul/pull/11576#discussion_r773218295)